updateStableIds/src/main/java/org/reactome/release/updateStableIds/UpdateStableIds.java

```diff
@@ -0,0 +1,171 @@
+				
+				// Compare number of 'Modified' instances between slices
+				Collection<GKInstance> sliceInstanceModified = sliceInstance.getAttributeValuesList(modified);
+				Collection<GKInstance> prevSliceInstanceModified = prevSliceInstance.getAttributeValuesList(modified);
+				if (sliceInstanceModified.size() > prevSliceInstanceModified.size()) {
+					// Make sure StableIdentifier instance exists
+					if (sliceInstance.getAttributeValue(stableIdentifier) != null && gkCentralInstance.getAttributeValue(stableIdentifier) != null) {
+						logger.info("\tIncrementing " + sliceInstance.getAttributeValue(stableIdentifier));
+						incrementStableIdentifier(sliceInstance, dbaSlice, sliceIE);
+						incrementStableIdentifier(gkCentralInstance, dbaGkCentral, gkCentralIE);
+						incrementedCount++;
+					} else if (sliceInstance.getAttributeValue(stableIdentifier) == null){
+						logger.warn(sliceInstance + ": could not locate StableIdentifier instance");
+					} else {
+						logger.warn(prevSliceInstance + ": Instance from previous slice did not have StableIdentifier instance");
+					}
+				} else {
+					if (sliceInstanceModified.size() < prevSliceInstanceModified.size()) {
+						logger.warn("Unexpected: Number of 'Modified' instances in [" + sliceInstance.getDBID() + "] is fewer than in previous release");
+					}
+					notIncrementedCount++;
+				}
+				// Instances that have been updated already during the current release will have their 'releaseStatus' attribute equal to 'UPDATED'.
+				// This will make sure that StableIDs are only updated once per release.
+				try {
+					if (isUpdated(sliceInstance, prevSliceInstance, dbaPrevSlice)) {
+						String releaseStatusString = (String) sliceInstance.getAttributeValue(releaseStatus);
+						String updated = "UPDATED";
+
+						if (releaseStatusString == null || !releaseStatusString.equals(updated)) {
+							sliceInstance.addAttributeValue(releaseStatus, updated);
+							sliceInstance.addAttributeValue(modified, sliceIE);
+							dbaSlice.updateInstanceAttribute(sliceInstance, releaseStatus);
+							dbaSlice.updateInstanceAttribute(sliceInstance, modified);
+						} else {
+							logger.info("StableIdentifer has already been updated during this release");
+						}
+					}
+				} catch (StackOverflowError e) {
+					e.printStackTrace();
+				}
+			} else if (gkCentralInstance == null) {
+				logger.warn(sliceInstance + " -- Instance not found in gkCentral");
+			}
+		}
+
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.commit();
+		}
+		logger.info("Commiting all changes in " + dbaGkCentral.getDBName());
+		dbaGkCentral.commit();
+		logger.info(incrementedCount + " Stable Identifiers were updated");
+		logger.info(notIncrementedCount + "were not updated");
+		logger.info("UpdateStableIdentifiers step has finished");
+		System.out.println(incrementedCount);
+	}
+
+	// Increments the identifierVersion attribute and updates the StableIdentifier displayName accordingly. TODO: IT
```

Comment:
The "IT" in "TODO: IT" - Is that for "Integration Test"?

Reply:
yes ahah, not very clear is it